### PR TITLE
Add subgraph sync status to navbar

### PR DESCRIPTION
**Pull request by bsamuels453**
Some strategists have run into issues where the hosted subgraph is syncing extraordinary slowly. 

On the Fantom network, the hosted subgraph can often be hundreds of blocks behind the tip of the chain. This leads to data consistency issues where strategists aren't sure if they've correctly migrated a strategy or if the subgraph is behind.

This pull request adds an indicator under the network selection menu that informs the user how far behind the subgraph is, if at all.

https://github.com/yearn/yearn-vaults-v2-subgraph/issues/150
https://github.com/yearn/yearn-watch/issues/211

this PR also fixes some memorization issues where memoized functions would cache data incorrectly across networks

### CODE REVIEW

File: src/components/app/NavBar/SyncStatus.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                {!subgraphRequestError &&
+                    !rpcRequestError &&
+                    subgraphData &&
+                    rpcData && (
+                        <Typography variant="caption" align="right">
+                            Subgraph is{' '}
+                            {rpcData.syncHeight
+                                .sub(subgraphData.syncHeight)
+                                .toString()}{' '}
+                            blocks behind
```
Comment (edwinzhng) [inline src/components/app/NavBar/SyncStatus.tsx]: small nit: I noticed ETH is generally only 1 block behind, so maybe `blocks` should be `block(s)` or we could drop the `s` if the sync height is equal to 1

File: src/utils/reports.ts [hunk 2]
```diff
@@ -240,4 +240,8 @@ export const _getReportsForStrategies = async (
     }
 };
 
-export const getReportsForStrategies = memoize(_getReportsForStrategies);
+// Functions with more than 2 parameters need a custom key defined for memoization to work correctly.
```
Comment (edwinzhng) [inline src/utils/reports.ts:243]: thanks for fixing these!

**Comment by bsamuels453** [comment]
@edwinzhng ready for re-review

**Comment by edwinzhng** [comment]
Screen Shot 2022-03-13 at 6 38 14 PM

Looks good, but somehow I'm seeing this negative number for Arbitrum, maybe some weird overflow error? I'll let @magic990619 do the final review and merge this though

**Comment by bsamuels453** [comment]
> Screen Shot 2022-03-13 at 6 38 14 PM
> 
> Looks good, but somehow I'm seeing this negative number for Arbitrum, maybe some weird overflow error? I'll let @magic990619 do the final review and merge this though

Unfortunately,  some node failure conditions will return a block height of 0 instead of an error. I know the error message is working right for preview (since its a cors error, ez to reproduce), but when it comes to nodes that bug out a lot (rpc.ftm.tools), it's much harder to reproduce.

**Review by edwinzhng — APPROVED** [review]

**Review by storming0x — APPROVED** [review]
LGTM!

**Pull request merged**